pkg/apis/eventing/v1beta2/eventtype_conversion.go

```diff
@@ -18,17 +18,86 @@ package v1beta2
 
 import (
 	"context"
-	"fmt"
 
 	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+
+	eventing "knative.dev/eventing/pkg/apis/eventing/v1"
+	"knative.dev/eventing/pkg/apis/eventing/v1beta3"
 )
 
-// ConvertTo implements apis.Convertible
+// ConvertTo converts the receiver into `to`.
 func (source *EventType) ConvertTo(ctx context.Context, to apis.Convertible) error {
-	return fmt.Errorf("v1beta2 is the highest known version, got: %T", to)
+	switch sink := to.(type) {
+	case *v1beta3.EventType:
+
+		source.ObjectMeta.DeepCopyInto(&sink.ObjectMeta)
+		source.Status.Status.DeepCopyInto(&sink.Status.Status)
+
+		sink.Spec.Reference = source.Spec.Reference.DeepCopy()
+		sink.Spec.Description = source.Spec.Description
+
+		if source.Spec.Reference == nil && source.Spec.Broker != "" {
+			source.Spec.Reference = &duckv1.KReference{
+				Kind:       "Broker",
+				Name:       source.Spec.Broker,
+				APIVersion: eventing.SchemeGroupVersion.String(),
+			}
+		}
+
+		sink.Spec.Attributes = []v1beta3.EventAttributeDefinition{}
+		if source.Spec.Type != "" {
+			sink.Spec.Attributes = append(sink.Spec.Attributes, v1beta3.EventAttributeDefinition{
+				Name:     "type",
+				Required: true,
+				Value:    source.Spec.Type,
+			})
+		}
+		if source.Spec.Schema != nil {
+			sink.Spec.Attributes = append(sink.Spec.Attributes, v1beta3.EventAttributeDefinition{
+				Name:     "schemadata",
+				Required: false,
+				Value:    source.Spec.Schema.String(),
+			})
+		}
```

Comment:
```suggestion
		if source.Spec.Schema != nil {
			sink.Spec.Attributes = append(sink.Spec.Attributes, v1beta3.EventAttributeDefinition{
				Name:     "schema",
				Required: false,
				Value:    source.Spec.Schema.String(),
			})
		}
```

Shouldn't this be something like `"schema"` or `"schemauri"` since the `SchemaData` is a different field on the eventtype v1beta2 spec? Afaik we probably want to convert both so we don't lose anything...

Reply:
The CE attribute is schemadata (the URL), the embedded schema is removed in v1b3 and it will not be converted

Reply:
@pierDipi does this mean we will have a lossy conversion? (I don't know if we actually guarantee lossless conversions for a beta api...)